crypto/engine/eng_devcrypto.c

```diff
@@ -53,66 +75,101 @@ struct cipher_ctx {
     int op;                      /* COP_ENCRYPT or COP_DECRYPT */
 };
 
-static const struct cipher_data_st {
+static struct cipher_data_st {
     int nid;
     int blocksize;
     int keylen;
     int ivlen;
     int flags;
     int devcryptoid;
+    int status;
+    int accelerated;
+    int enabled;
+    char *driver_name;
```

Comment:
I've been wondering for a time if these shouldn't be in a different structure and array.  The benefit would be that the array of states could be easily copied and restored...  see my comments in the control command function.

Reply:
Unlike .enabled, I think the other additions are OK here.  It doesn't really matter to me.  If you want me to move them to a `struct driver_info` table, I will.

Reply:
I thought about it as well, and I am making the move.  It will make it cleaner.